src/ArborX_DistributedSearchTree.hpp

```diff
@@ -117,12 +119,11 @@ DistributedSearchTree<MemorySpace, Enable>::DistributedSearchTree(
 
   // Create new context for the library to isolate library's communication from
   // user's
-  auto *newcomm_ptr = new MPI_Comm();
-  MPI_Comm_dup(comm, newcomm_ptr);
-  _comm_ptr.reset(newcomm_ptr, [](MPI_Comm *p) {
+  _comm_ptr.reset(new MPI_Comm(), [](MPI_Comm *p) {
```

Comment:
I meant to implement it the way I did.  What's the rational for your change?

Reply:
My rational was if the destructor somehow is called before `MPI_Comm_dup()` is called we cannot recover from it.

Reply:
I'd be fine with `reset([comm](){ auto p = new MPI_Comm(); MPI_Comm_dup(comm, p); return p; }(), ...)`

Reply:
I like the latest variant. Please push it.